c7n/resources/s3.py

```diff
@@ -3707,3 +3708,101 @@ def process_bucket(self, b):
                 raise
             controls = {}
         b[self.annotation_key] = controls.get('OwnershipControls')
+
+
+@filters.register('bucket-replication')
+class BucketReplication(Filter):
+    """Filters for S3 buckets that have replication-rule
+
+    Reference: https://docs.aws.amazon.com/AmazonS3/latest/userguide/replication.html
+
+    :example
+
+    Find buckets with replication rule enabled
```

Comment:
how does this handle cross account logging?

it also feels like we would potentially want to expose the ability to look at the rule not merely assert a destination on it, ie only replicating a particular path would satisfy here. and for the destination assertion potentially assert an actual value not just the symbolics on cross-region/same-region.

Reply:
Fair point about being able to check all the settings in one spot. Are you thinking something like augmenting `Replication.Rules[]` with destination bucket region info and exposing the whole list as a `ListItemFilter`?

Reply:
Yeah list-item on rules would address, the cross-account symbolic might also be interesting (and also verifies that its resilient in the case of that)

Reply:
yeah, good suggestion. I will update this to use ListItemFilter.